src/Assets/Javascript/map.js

```diff
@@ -1,4 +1,114 @@
+const createPopupClass = () => {
+  const panToWithOffset = function(map, latlng, offsetX, offsetY) {
+    const ov = new google.maps.OverlayView()
+    ov.onAdd = function() {
+      const proj = this.getProjection()
+      const aPoint = proj.fromLatLngToContainerPixel(latlng)
+      aPoint.x = aPoint.x + offsetX
+      aPoint.y = aPoint.y + offsetY
+      map.panTo(proj.fromContainerPixelToLatLng(aPoint))
+    }
+    ov.draw = function() {}
+    ov.setMap(map)
+  }
+
+  // Based on: https://developers.google.com/maps/documentation/javascript/examples/overlay-popup
+  /**
+   * A customized popup on the map.
+   * @param {!google.maps.LatLng} position
+   * @param {!Element} content
+   * @constructor
+   * @extends {google.maps.OverlayView}
+   */
+  const Popup = function(position, closedContent, openContent) {
+    this.position = position
+    const content = document.createElement("div")
+    content.classList.add("popup-bubble-content")
+    closedContent.classList.add("popup-bubble__closed-content")
+    openContent.classList.add("popup-bubble__open-content")
+    content.appendChild(closedContent)
+    content.appendChild(openContent)
+
+    var pixelOffset = document.createElement("div")
+    pixelOffset.classList.add("popup-bubble-anchor")
+    pixelOffset.appendChild(content)
+
+    this.anchor = document.createElement("div")
+    this.anchor.classList.add("popup-tip-anchor")
+    this.anchor.appendChild(pixelOffset)
+
+    // Optionally stop clicks, etc., from bubbling up to the map.
+    this.stopEventPropagation()
+
+    closedContent.addEventListener("click", e => {
+      panToWithOffset(this.getMap(), this.position, 0, -150)
+      this.closeOpenPopups()
+      this.anchor.classList.toggle("open")
+      e.stopPropagation()
+    })
+
+    const $closeButton = openContent.querySelector(".popup-bubble__close")
+    $closeButton.addEventListener("click", e => {
+      this.closeOpenPopups()
```

Comment:
missing ` e.stopPropagation()`

Reply:
Shouldn't be needed because the element underneath has stopPropagation set up on it.